### PR TITLE
fix: bundle lambdas in script instead of command line

### PR DESCRIPTION
**Pull request by ssvegaraju**
Issue #, if available:

Description of changes:
Created a script to move files as needed instead of using command line hooks. This resolves the failing action here https://github.com/awslabs/fhir-works-on-aws-deployment/actions/runs/2678017744

Checklist:

* [X] Have you successfully deployed to an AWS account with your changes?
* [X] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/build_lambda.js [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+    }
+    fs.mkdirSync(dirname, { recursive: true });
+}
+
+if (isDirectory) {
+    fse.copySync(inputDir, outputDir);
+} else {
+    ensureDirectoryExistence(`${outputDir}\\${fileToMove}`);
+    fs.copyFileSync(`${inputDir}\\${fileToMove}`, `${outputDir}\\${fileToMove}`);
+}
```
Comment (Bingjiling) [inline scripts/build_lambda.js:26]: nit - No new line at end of file

**Review by Bingjiling — DISMISSED** [review]
Looks good! Could you please add the link of the failed GitHub action run to provide context for this fix?

File: src/subscriptions/restHookLambda/restHook.ts [hunk 2]
```diff
@@ -67,8 +66,9 @@ export default class RestHookHandler {
         event: SQSEvent,
         allowListPromise: Promise<{ [key: string]: AllowListInfo }>,
     ): Promise<SQSBatchResponse> {
-        await ensureAsyncInit(allowListPromise);
+        console.log(allowListPromise);
```
Comment (Bingjiling) [inline src/subscriptions/restHookLambda/restHook.ts]: Use `logger.debug` instead? The allowList might contain sensitive information that customers do not want in the log all the time.

**Review by Bingjiling — COMMENTED** [review]
Looks good! One last thing on logging -

**Review by Bingjiling — APPROVED** [review]

File: src/subscriptions/restHookLambda/restHook.ts [hunk 3]
```diff
@@ -67,8 +66,9 @@ export default class RestHookHandler {
         event: SQSEvent,
         allowListPromise: Promise<{ [key: string]: AllowListInfo }>,
     ): Promise<SQSBatchResponse> {
-        await ensureAsyncInit(allowListPromise);
+        logger.debug(allowListPromise);
         const allowList = await allowListPromise;
+        logger.debug(allowList);
         const messages = event.Records.map((record: any): SubscriptionNotification => {
```
Comment (rsmayda) [inline src/subscriptions/restHookLambda/restHook.ts:72]: looks like you are no longer calling 
```ts
        await ensureAsyncInit(allowListPromise);
```
did you mean for that?

File: lib/cdk-infra-stack.ts [hunk 4]
```diff
@@ -381,12 +389,10 @@ export default class FhirWorksStack extends Stack {
...
                         // this allows the lambda functions for bulk export to have access to these files within the lambda instance
                         return [
-                            `dir ${outputDir}\\bulkExport || mkdir -p ${outputDir}\\bulkExport\\glueScripts`,
-                            `dir ${outputDir}\\bulkExport\\schema || mkdir ${outputDir}\\bulkExport\\schema`,
-                            `cp ${inputDir}\\bulkExport\\glueScripts\\export-script.py ${outputDir}\\bulkExport\\glueScripts\\export-script.py`,
-                            `cp ${inputDir}\\bulkExport\\schema\\transitiveReferenceParams.json ${outputDir}\\bulkExport\\schema\\transitiveReferenceParams.json`,
-                            `cp ${inputDir}\\bulkExport\\schema\\${PATIENT_COMPARTMENT_V3} ${outputDir}\\bulkExport\\schema\\${PATIENT_COMPARTMENT_V3}`,
-                            `cp ${inputDir}\\bulkExport\\schema\\${PATIENT_COMPARTMENT_V4} ${outputDir}\\bulkExport\\schema\\${PATIENT_COMPARTMENT_V4}`,
+                            `node scripts/build_lambda.js ${inputDir} ${outputDir} bulkExport\\glueScripts\\export-script.py`,
```
Comment (rsmayda) [inline lib/cdk-infra-stack.ts:392]: I think we will want forward slashes not back slashes - https://stackoverflow.com/a/38428899

**Pull request merged**